diff --git a/hw_06/clib/src/common/clib-package.c b/hw_06/clib/src/common/clib-package.c
index 268538e..ee3570b 100644
--- a/hw_06/clib/src/common/clib-package.c
+++ b/hw_06/clib/src/common/clib-package.c
@@ -654,6 +654,9 @@ clib_package_new_from_slug_with_package_name(const char *slug, int verbose,
     if (retries-- <= 0) {
       goto error;
     } else {
+      if(res) {
+        http_get_free(res);
+      }
 #ifdef HAVE_PTHREADS
       init_curl_share();
       _debug("GET %s", json_url);
